addon/-private/class/modifier.ts

```diff
@@ -261,4 +295,71 @@ export default class ClassBasedModifier<S = DefaultSignature> {
   }
 }
 
+// We apply these here, against the prototype, so that there is only one of
+// these, rather than one per instance. We also only issue the deprecation once
+// per class for each of `args` and `element`.
+Object.defineProperty(ClassBasedModifier.prototype, 'args', {
+  enumerable: true,
+  get(this: InternalClassBasedModifier<unknown>) {
+    deprecate(
+      `ember-modifier (in ${this.constructor.name} at ${
+        new Error().stack
+      }): using \`this.args\` is deprecated. Access positional and named arguments directly in the \`modify\` hook instead.`,
+      !SEEN_CLASSES_FOR_ARGS.has(this.constructor),
```

Comment:
I think we should also avoid the deprecation for this case:

https://github.com/ember-modifier/ember-modifier/blob/d277c490a4b5ae3d03df29443095b8eeac5ae294/addon/-private/class/modifier.ts#L111

Specifically, we should try to avoid emitting a deprecation for needing to migrate to `.modify` **and** accessing `args`

Reply:
chatted with @chriskrycho, this is still helpful for folks doing smaller transitions (e.g. rename `didReceiveArguments` to `modify` direclty), so they still see the other deprecated things to clean up...